catatonit.c

```diff
@@ -136,6 +139,81 @@ int kernel_signals[] = {SIGFPE, SIGILL, SIGSEGV, SIGBUS, SIGABRT, SIGTRAP, SIGSY
 
 #define ARRAY_LEN(arr) (sizeof(arr) / sizeof(*arr))
 
+static int syscall_close_range(unsigned int fd, unsigned int max_fd, unsigned int flags)
+{
+#ifdef __NR_close_range
+  return (int) syscall(__NR_close_range, fd, max_fd, flags);
+#else
+  errno = ENOSYS;
+  return -1;
+#endif
+}
+
+/*
+ * Close every fd >= n that is different from exclude_fd using close_range.
+ */
+static int close_range_fds_ge_than(int n, int exclude_fd)
+{
+	int r;
+
+        /* exclude_fd is not in the [n, UINT_MAX] range.  */
+	if (exclude_fd < n)
+		return syscall_close_range(n, UINT_MAX, 0);
+
+        /* exclude_fd is the first fd in the [n, UINT_MAX] range.  */
+	if (exclude_fd == n)
+		return syscall_close_range(n + 1, UINT_MAX, 0);
+
+        /* exclude_fd is between n and UINT_MAX.  */
+        r = syscall_close_range(n, exclude_fd - 1, 0);
+	if (r < 0)
+		return r;
+        if (exclude_fd < UINT_MAX)
+		return syscall_close_range(exclude_fd + 1, UINT_MAX, 0);
+
+          return 0;
+}
```

Comment:
codewise this looks ok, but the indentation seems off to me, or is it just the github ui?